pkg/apis/options/legacy_options.go

```diff
@@ -115,3 +128,85 @@ func (l *LegacyUpstreams) convert() (Upstreams, error) {
 
 	return upstreams, nil
 }
+
+// LegacyAuthorization holds configuration options related to trusted requests which
+// would skip authentication
+type LegacyAuthorization struct {
+	SkipAuthRegex     []string `flag:"skip-auth-regex" cfg:"skip_auth_regex"`
+	SkipAuthRoutes    []string `flag:"skip-auth-route" cfg:"skip_auth_routes"`
+	SkipAuthPreflight bool     `flag:"skip-auth-preflight" cfg:"skip_auth_preflight"`
+	TrustedIPs        []string `flag:"trusted-ip" cfg:"trusted_ips"`
+}
+
+// legacyAuthorizationFlagSet creates a new FlagSet with all of the flags required by Authorization
+func legacyAuthorizationFlagSet() *pflag.FlagSet {
+	flagSet := pflag.NewFlagSet("authorization", pflag.ExitOnError)
+
+	flagSet.StringSlice("skip-auth-regex", []string{}, "(DEPRECATED for --skip-auth-route) bypass authentication for requests path's that match (may be given multiple times)")
+	flagSet.StringSlice("skip-auth-route", []string{}, "bypass authentication for requests that match the method & path. Format: method=path_regex OR path_regex alone for all methods")
+	flagSet.Bool("skip-auth-preflight", false, "will skip authentication for OPTIONS requests")
+	flagSet.StringSlice("trusted-ip", []string{}, "list of IPs or CIDR ranges to allow to bypass authentication. WARNING: trusting by IP has inherent security flaws, read the configuration documentation for more information.")
+
+	return flagSet
+}
+
+func (l *LegacyAuthorization) convert() RequestRules {
+	var rules RequestRules
+
+	l.convertRoutes(rules)
+	l.convertRegexes(rules)
+	l.convertPreflight(rules)
+	l.convertTrustedIPs(rules)
```

Comment:
Probably better to return a list of rules and append them rather than mutating an argument

eg `rules = append(rules, l.convertRoutes()...)`